diff --git a/comm/suite/installer/package-manifest.in b/comm/suite/installer/package-manifest.in
index 0b8f4d8..7e815f4 100644
--- a/comm/suite/installer/package-manifest.in
+++ b/comm/suite/installer/package-manifest.in
@@ -604,6 +604,7 @@
 
 ; [Default Preferences]
 ; All the browser/general pref files must be part of base to prevent migration bugs
+@RESPATH@/@PREF_DIR@/all-gentoo.js
 @RESPATH@/@PREF_DIR@/suite-prefs.js
 @RESPATH@/@PREF_DIR@/composer.js
 @RESPATH@/greprefs.js
