--- seamonkey-2.53.6b1/comm/suite/installer/package-manifest.in
+++ seamonkey-2.53.6b1/comm/suite/installer/package-manifest.in
@@ -603,16 +603,17 @@
 @RESPATH@/chrome/icons/default/places.ico
 @RESPATH@/chrome/icons/default/script-file.ico
 @RESPATH@/chrome/icons/default/xml-file.ico
 @RESPATH@/chrome/icons/default/xul-file.ico
 #endif
 
 ; [Default Preferences]
 ; All the browser/general pref files must be part of base to prevent migration bugs
+@RESPATH@/@PREF_DIR@/all-gentoo.js
 @RESPATH@/@PREF_DIR@/suite-prefs.js
 @RESPATH@/@PREF_DIR@/composer.js
 @RESPATH@/greprefs.js
 @RESPATH@/defaults/autoconfig/prefcalls.js
 @RESPATH@/defaults/permissions
 @RESPATH@/defaults/blocklists
 @RESPATH@/defaults/pinning
 ; Warning: changing the path to channel-prefs.js can cause bugs. (Bug 756325)
