From bdd6d60f76aa7a6c0bdbc2a190bb8c9493cd6e2d Mon Sep 17 00:00:00 2001
From: Marcus Comstedt <marcus@mc.pp.se>
Date: Sat, 2 Apr 2022 11:59:41 +0200
Subject: [PATCH] Don't use JIT signal handlers if no JIT

---
 js/src/wasm/WasmSignalHandlers.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/js/src/wasm/WasmSignalHandlers.cpp.orig b/js/src/wasm/WasmSignalHandlers.cpp
index 20d0268..d4fee57 100644
--- a/js/src/wasm/WasmSignalHandlers.cpp.orig
+++ b/js/src/wasm/WasmSignalHandlers.cpp
@@ -1691,6 +1691,8 @@ ProcessHasSignalHandlers()
 bool
 wasm::EnsureSignalHandlers(JSContext* cx)
 {
+#ifndef JS_CODEGEN_NONE
+
     // Nothing to do if the platform doesn't support it.
     if (!ProcessHasSignalHandlers())
         return true;
@@ -1699,6 +1701,8 @@ wasm::EnsureSignalHandlers(JSContext* cx)
     // On OSX, each JSContext which runs wasm gets its own handler thread.
     if (!cx->wasmMachExceptionHandler.installed() && !cx->wasmMachExceptionHandler.install(cx))
         return false;
+#endif
+
 #endif
 
     return true;
-- 
2.35.1

