diff -urN seamonkey-2.53.14.orig/config/system-headers.mozbuild seamonkey-2.53.14/config/system-headers.mozbuild
--- seamonkey-2.53.14.orig/config/system-headers.mozbuild	2020-12-22 12:42:29.000000000 +0100
+++ seamonkey-2.53.14/config/system-headers.mozbuild	2022-09-30 23:01:10.397647428 +0200
@@ -1349,3 +1349,10 @@
         'unicode/ustring.h',
         'unicode/utypes.h',
     ]
+
+if CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    system_headers += [
+	'harfbuzz/hb-glib.h',
+	'harfbuzz/hb-ot.h',
+	'harfbuzz/hb.h'
+    ]
\ Geen regeleindeteken (LF) aan einde van bestand
diff -urN seamonkey-2.53.14.orig/dom/base/moz.build seamonkey-2.53.14/dom/base/moz.build
--- seamonkey-2.53.14.orig/dom/base/moz.build	2022-08-24 21:55:10.000000000 +0200
+++ seamonkey-2.53.14/dom/base/moz.build	2022-09-30 23:02:06.879047095 +0200
@@ -476,6 +476,9 @@
 if CONFIG['MOZ_X11']:
     CXXFLAGS += CONFIG['TK_CFLAGS']
 
+if CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    CXXFLAGS += CONFIG['MOZ_HARFBUZZ_CFLAGS']
+
 GeneratedFile('UseCounterList.h', script='gen-usecounters.py',
               entry_point='use_counter_list', inputs=['UseCounters.conf'])
 
diff -urN seamonkey-2.53.14.orig/gfx/harfbuzz/README-mozilla seamonkey-2.53.14/gfx/harfbuzz/README-mozilla
--- seamonkey-2.53.14.orig/gfx/harfbuzz/README-mozilla	2021-10-26 18:49:54.000000000 +0200
+++ seamonkey-2.53.14/gfx/harfbuzz/README-mozilla	2022-09-30 23:03:01.089350281 +0200
@@ -21,3 +21,8 @@
 
 If the collection of source files changes, manual updates to moz.build may be
 needed as we don't use the upstream makefiles.
+
+The in-tree copy may be omitted during build by --with-system-harfbuzz.
+Make sure to keep pkg-config version check within toolkit/moz.configure in sync
+with checkout version or increment latest tag by one if it's not based
+on upstream release.
\ Geen regeleindeteken (LF) aan einde van bestand
diff -urN seamonkey-2.53.14.orig/gfx/moz.build seamonkey-2.53.14/gfx/moz.build
--- seamonkey-2.53.14.orig/gfx/moz.build	2020-02-18 00:37:51.000000000 +0100
+++ seamonkey-2.53.14/gfx/moz.build	2022-09-30 23:03:55.458660216 +0200
@@ -10,6 +10,9 @@
 if CONFIG['MOZ_TREE_CAIRO']:
     DIRS += ['cairo']
 
+if not CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    DIRS += ['harfbuzz/src']
+
 DIRS += [
     '2d',
     'ycbcr',
@@ -19,7 +22,6 @@
     'gl',
     'layers',
     'graphite2/src',
-    'harfbuzz/src',
     'ots/src',
     'thebes',
     'ipc',
diff -urN seamonkey-2.53.14.orig/gfx/skia/generate_mozbuild.py seamonkey-2.53.14/gfx/skia/generate_mozbuild.py
--- seamonkey-2.53.14.orig/gfx/skia/generate_mozbuild.py	2022-08-24 21:55:11.000000000 +0200
+++ seamonkey-2.53.14/gfx/skia/generate_mozbuild.py	2022-09-30 23:05:18.557190750 +0200
@@ -149,6 +149,9 @@
         '-Wno-unused-private-field',
     ]
 
+if CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    CXXFLAGS += CONFIG['MOZ_HARFBUZZ_CFLAGS']
+
 if CONFIG['MOZ_WIDGET_TOOLKIT'] in ('gtk3', 'android'):
     CXXFLAGS += CONFIG['MOZ_CAIRO_CFLAGS']
     CXXFLAGS += CONFIG['CAIRO_FT_CFLAGS']
diff -urN seamonkey-2.53.14.orig/gfx/thebes/moz.build seamonkey-2.53.14/gfx/thebes/moz.build
--- seamonkey-2.53.14.orig/gfx/thebes/moz.build	2022-08-24 21:55:11.000000000 +0200
+++ seamonkey-2.53.14/gfx/thebes/moz.build	2022-09-30 23:06:40.424668971 +0200
@@ -269,6 +269,9 @@
 
 DEFINES['GRAPHITE2_STATIC'] = True
 
+if CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    CXXFLAGS += CONFIG['MOZ_HARFBUZZ_CFLAGS']
+
 if CONFIG['CC_TYPE'] == 'clang':
     # Suppress warnings from Skia header files.
     SOURCES['gfxPlatform.cpp'].flags += ['-Wno-implicit-fallthrough']
diff -urN seamonkey-2.53.14.orig/intl/unicharutil/util/moz.build seamonkey-2.53.14/intl/unicharutil/util/moz.build
--- seamonkey-2.53.14.orig/intl/unicharutil/util/moz.build	2020-08-10 13:30:34.000000000 +0200
+++ seamonkey-2.53.14/intl/unicharutil/util/moz.build	2022-09-30 23:07:32.146866442 +0200
@@ -25,4 +25,7 @@
     'nsUnicodeProperties.cpp',
 ]
 
+if CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    CXXFLAGS += CONFIG['MOZ_HARFBUZZ_CFLAGS']
+
 FINAL_LIBRARY = 'xul'
diff -urN seamonkey-2.53.14.orig/netwerk/dns/moz.build seamonkey-2.53.14/netwerk/dns/moz.build
--- seamonkey-2.53.14.orig/netwerk/dns/moz.build	2022-08-24 21:55:13.000000000 +0200
+++ seamonkey-2.53.14/netwerk/dns/moz.build	2022-09-30 23:08:27.444215814 +0200
@@ -65,6 +65,9 @@
     '/netwerk/base',
 ]
 
+if CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    CXXFLAGS += CONFIG['MOZ_HARFBUZZ_CFLAGS']
+
 USE_LIBS += ['icu']
 
 if CONFIG['CC_TYPE'] in ('clang', 'gcc'):
diff -urN seamonkey-2.53.14.orig/toolkit/library/moz.build seamonkey-2.53.14/toolkit/library/moz.build
--- seamonkey-2.53.14.orig/toolkit/library/moz.build	2022-08-24 21:55:14.000000000 +0200
+++ seamonkey-2.53.14/toolkit/library/moz.build	2022-09-30 23:09:30.181881283 +0200
@@ -225,6 +225,9 @@
 if CONFIG['MOZ_SYSTEM_PNG']:
     OS_LIBS += CONFIG['MOZ_PNG_LIBS']
 
+if CONFIG['MOZ_SYSTEM_HARFBUZZ']:
+    OS_LIBS += CONFIG['MOZ_HARFBUZZ_LIBS']
+
 if CONFIG['MOZ_SYSTEM_WEBP']:
     OS_LIBS += CONFIG['MOZ_WEBP_LIBS']
 
diff -urN seamonkey-2.53.14.orig/toolkit/moz.configure seamonkey-2.53.14/toolkit/moz.configure
--- seamonkey-2.53.14.orig/toolkit/moz.configure	2022-08-24 21:55:14.000000000 +0200
+++ seamonkey-2.53.14/toolkit/moz.configure	2022-09-30 23:10:51.929354414 +0200
@@ -892,6 +892,21 @@
 add_old_configure_assignment('FT2_CFLAGS',
                              ft2_info.cflags)
 
+# HarfBuzz
+# ==============================================================
+option('--with-system-harfbuzz',
+       help='Use system harfbuzz (located with pkgconfig)')
+
+@depends('--with-system-harfbuzz')
+def check_for_harfbuzz(value):
+    return bool(value)
+
+system_harfbuzz = pkg_check_modules('MOZ_HARFBUZZ', 'harfbuzz >= 1.3.3',
+				    when=check_for_harfbuzz)
+
+set_config('MOZ_SYSTEM_HARFBUZZ', depends_if(system_harfbuzz)(lambda _: True))
+
+
 # Mortar
 # ==============================================================
 option('--enable-mortar', help='Enable mortar extension')
