diff --git a/build/moz.configure/init.configure b/build/moz.configure/init.configure
index 7e1eed4..e921e3b 100644
--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -310,6 +310,8 @@ def virtualenv_python3(env_python, build_env, mozconfig, help):
     try:
         import mozfile
     except ImportError:
+        sys.path.insert(
+            0, os.path.join(topsrcdir, 'testing', 'mozbase', 'mozfile'))
         sys.path.insert(
             0, os.path.join(topsrcdir, 'testing', 'mozbase', 'mozfile'))
         sys.path.insert(
