diff -urN seamonkey-2.53.14.orig/build/moz.configure/init.configure seamonkey-2.53.14/build/moz.configure/init.configure
--- seamonkey-2.53.14.orig/build/moz.configure/init.configure	2022-08-24 21:55:10.000000000 +0200
+++ seamonkey-2.53.14/build/moz.configure/init.configure	2022-10-02 11:19:56.594415178 +0200
@@ -267,11 +267,6 @@
             die('The PYTHON environment variable does not contain '
                 'a valid path. Cannot find %s', python)
         python = found_python
-        try:
-            version = python_executable_version(python).version
-        except Exception as e:
-            raise FatalCheckError(
-                'could not determine version of given PYTHON: %s' % e)
     else:
         python = sys.executable
 
